### DISCUSSION TITLE
project Defi stable coin : invariant testing

### DISCUSSION THREAD

**Discussion by ighodaro-daniel** [Cyfrin/foundry-full-course-cu#2305] [Q&A]
i am trying to test the mintdsc using  invariant but keep getting this error: Encountered 1 failing test in test/fuzz/VariantTest.t.sol:OpenInvariantTest
[FAIL. Reason: invariant_protocolMustHaveMoreValueThanTotalSupply replay failure]
        [Sequence]
                sender=0x0000000000000000000000000000000000000517 addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b calldata=depositCollateral(uint256,uint256) args=[791, 1086394137 [1.086e9]]
                sender=0x00000000000000000000000000000000194f1398 addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b calldata=mint(uint256,uint256) args=[11038080538831016375826685593578567593976028506259816990278630323878396837 [1.103e73], 12740058083786827124630583 [1.274e25]]
                sender=0x0000000000000000000000000000000000000394 addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b calldata=mint(uint256,uint256) args=[238642156304249311964020505196285433545111626934982994721 [2.386e56], 38589232168242703 [3.858e16]]
                sender=0x00000000000000000000000000000000E90dB8A4 addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b calldata=mint(uint256,uint256) args=[744, 2459]
                sender=0x0000000000000000000000000000000040c10F19 addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b calldata=mint(uint256,uint256) args=[123, 622]
                sender=0x0000000000000000000000000000000000001252 addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b calldata=depositCollateral(uint256,uint256) args=[5796867799043533175782 [5.796e21], 10380980367096386810822019524423620913185047670980114606872230465897277 [1.038e70]]
                sender=0x000000000000000000000000000000000000092F addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b calldata=depositCollateral(uint256,uint256) args=[21, 989]
                sender=0x0000000000000000000000000000000000000517 addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b calldata=depositCollateral(uint256,uint256) args=[351, 80771158706477507628043956924673312329319157808077710294928738145137779483126 [8.077e76]]
                sender=0x00000000000000000000000000000000000000D4 addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b calldata=mint(uint256,uint256) args=[661, 1527]

**Comment by ighodaro-daniel** [comment]
i have cross check the _getUsdValue function from the dscengine for any overflow or underflow error and my test passed it, please help in figuring out the issue

**Reply by EngrPips** [reply]
run `forge clean`, then `foundryup` then retry the test and share the output with us.

**Comment by ighodaro-daniel** [comment]
okay , i will do that right away

**Comment by ighodaro-daniel** [comment]
i have tried it  and still get this error: Ran 1 test for
test/fuzz/VariantTest.t.sol:OpenInvariantTest
[FAIL. Reason: panic: assertion failed (0x01)]
        [Sequence]
                sender=0x00000000000000000000000000000000000001bd
addr=[test/fuzz/HandlerTest.t.sol:HandlerTest]0x2e234DAe75C793f67A35089C9d99245E1C58470b
calldata=depositCollateral(uint256,uint256) args=[6,
3568751434479760106431390877299209589119333339702385792606112571173
[3.568e66]]

**Reply by EngrPips** [reply]
Hey @ighodaro-daniel, the test is failing with an assertion failed error, so this is a more sensible error, and you have to look through your code to find out why your assertion is incorrect.
